animation_nodes/base_types/tree_auto_execution.py

```diff
@@ -16,69 +14,96 @@ class AutoExecutionTrigger_MonitorProperty(bpy.types.PropertyGroup):
-        if hasattr(prop, "__iter__"):
-            return " ".join(str(part) for part in prop)
-        return str(prop)
+    def getPropertiesState(self):
+        props = self.getProperties()
+        if len(props) == 0: return ""
+
+        propsString = ""
+        for prop in props:
+            if hasattr(prop, "__iter__"):
+                propsString += "".join(str(part) for part in prop)
+            else: propsString += str(prop)
 
-    def getProperty(self):
+        return propsString
+
+    def getProperties(self):
         self.hasError = False
         object = self.getObject()
-        if object is None or self.dataPath is "":
-            return None
-
-        try: return object.path_resolve(self.dataPath)
+        if object is None or self.dataPaths.strip() is "":
+            return []
+
+        try:
+            paths = self.dataPaths.split(",")
+            if self.idType == "COLLECTION":
+                properties = []
+                for obj in object.all_objects:
+                    properties.extend(obj.path_resolve(p.strip()) for p in paths)
+                return properties
+            else:
+                return [object.path_resolve(p.strip()) for p in paths]
         except:
             self.hasError = True
-            return None
+            return []
 
     def getObject(self):
         if self.idType == "OBJECT":
             return self.object
+        elif self.idType == "COLLECTION":
+            return self.collection
         elif self.idType == "SCENE":
             return self.scene
 
     def draw(self, layout, index):
-        row = layout.row(align = True)
-        if self.hasError:
-            row.label(text = "", icon = "ERROR")
+        box = layout.box()
 
-        icon = "LAYER_ACTIVE" if self.enabled else "LAYER_USED"
-        row.prop(self, "enabled", icon = icon, text = "")
+        header = box.row()
+        icon = 'TRIA_DOWN' if self.expanded else 'TRIA_RIGHT'
+        header.prop(self, "expanded", icon = icon, text = "", emboss = False)
+        if self.hasError: header.label(text = "", icon = "ERROR")
 
-        row.active = self.enabled
+        enableText = "Enable "
+        object = self.getObject()
+        if object is not None: enableText += object.name + "." + self.dataPaths
+        header.prop(self, "enabled", text = enableText, toggle = True)
 
-        props = row.operator("an.assign_active_object_to_auto_execution_trigger", icon = "EYEDROPPER", text = "")
-        props.index = index
+        header.operator("an.remove_auto_execution_trigger", icon = "X",
+            text = "", emboss = False).index = index
 
-        if self.idType == "OBJECT":
-            row.prop(self, "object", text = "")
-        elif self.idType == "SCENE":
-            row.prop(self, "scene", text = "")
+        if self.expanded:
+            col = box.column(align = True)
+            col.active = self.enabled
+            col.prop(self, "idType", text = "")
```

Comment:
Maybe it would be better if it is impossible to change the `idType`.
Then the API can become cleaner and it is easier to avoid references to objects/scenes/collections that are not actually used by the trigger.

Reply:
What do you mean by "not actually used by the trigger"?

Reply:
I wanted to provide the enum directly because it is much faster to use, compared to deleting and adding again.

Reply:
For example the trigger could have a reference to a collection AND to a scene, although only one of those is used. This could become problematic when the node tree is appended into a new file because Blender will try to append the linked stuff as well (afaik).

How often do you actually change the type? Not sure if it is worth having this option in the UI.

Reply:
I see. So I should bring back the the pop up menu as before?

Reply:
An alternative would be to reset the other fields when the type changes. Whatever you prefer.

Reply:
I chose the later. I think it is more user friendly.